src/init/ClusterManager.ts

```diff
@@ -0,0 +1,120 @@
+import type { Worker } from 'cluster';
+import cluster from 'cluster';
+import { cpus } from 'os';
+import { getLoggerFor } from '../logging/LogUtil';
+import { InternalServerError } from '../util/errors/InternalServerError';
+
+/**
+ * Different cluster modes.
+ */
+enum ClusterMode {
+  /** Scales in relation to `core_count`. */
+  autoScale,
+  /** Single threaded mode, no clustering */
+  singleThreaded,
+  /** Fixed amount of workers being forked. (limited to core_count) */
+  fixed
+}
+
+/**
+ * Convert workers amount to {@link ClusterMode}
+ * @param workers - Amount of workers (typically passed through cli parameter)
+ * @returns ClusterMode enum value
+ */
+function toClusterMode(workers: number | string): ClusterMode {
```

Comment:
```suggestion
function toClusterMode(workers: number): ClusterMode {
```
Since parsing now happens in the constructor.